### DISCUSSION TITLE
Building for iOS

### DISCUSSION THREAD

**Discussion by exilef** [Tencent/libpag#2293] [Q&A] [answered]
Hi, I could not find documentation for how to build the libpag library for iOS on an Apple Silicon (M2) macOS machine. How does this work with either plain CMake or CLion?

**Comment by kevingpqi123** [comment, marked as answer]
The XCFramework we provide includes the arm64 simulator architecture, which can be used directly.

If you need to compile it yourself, you can refer to the README file in the root directory. Execute the relevant commands to obtain third-party library dependencies. Also, check the README file in the ios directory to generate projects for real devices or simulators. The remaining steps are standard compilation processes. By compiling the project files, you can generate frameworks for various architectures.
